[CI] AccountableQueryCircuitBreakerIT testRegexpQueryMemoryReleased failing

**Build Scans:**
- [elasticsearch-intake #40122 / part1](https://gradle-enterprise.elastic.co/s/jqusfficfsdsy)

**Reproduction Line:**
```
./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.indices.memory.breaker.AccountableQueryCircuitBreakerIT.testRegexpQueryMemoryReleased" -Dtests.seed=BE846BD61752503A -Dtests.locale=hi-IN -Dtests.timezone=America/Miquelon -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.indices.memory.breaker.AccountableQueryCircuitBreakerIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRegexpQueryMemoryReleased'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Request breaker memory should be released after search completes expected:<81920> but was:<0>
```

**Issue Reasons:**
- [main] 1 consecutive failures in step part1
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -570,9 +570,6 @@ tests:
 - class: org.elasticsearch.index.reindex.ResumeBulkByScrollRequestWireSerializingTests
   method: testEqualsAndHashcode
   issue: https://github.com/elastic/elasticsearch/issues/149721
-- class: org.elasticsearch.indices.memory.breaker.AccountableQueryCircuitBreakerIT
-  method: testRegexpQueryMemoryReleased
-  issue: https://github.com/elastic/elasticsearch/issues/149727
 - class: org.elasticsearch.xpack.stateless.StatelessCommitServiceIT
   method: testRetainCommitForReadersAfterShardReplicasSetToZero
   issue: https://github.com/elastic/elasticsearch/issues/149740

--- a/server/src/internalClusterTest/java/org/elasticsearch/indices/memory/breaker/AccountableQueryCircuitBreakerIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/indices/memory/breaker/AccountableQueryCircuitBreakerIT.java
@@ -253,11 +253,10 @@ public class AccountableQueryCircuitBreakerIT extends ESIntegTestCase {
     }
 
     private void assertQueryMemoryReleased(QueryBuilder query) throws Exception {
-        long baseline = getRequestBreakerEstimated();
+        assertBusy(() -> assertEquals("Request breaker should be empty before search", 0L, getRequestBreakerEstimated()));
         client().prepareSearch(INDEX_NAME).setQuery(query).get().decRef();
-        assertBusy(() -> {
-            long estimated = getRequestBreakerEstimated();
-            assertEquals("Request breaker memory should be released after search completes", baseline, estimated);
-        });
+        assertBusy(
+            () -> assertEquals("Request breaker memory should be released after search completes", 0L, getRequestBreakerEstimated())
+        );
     }
 }

